Private/Get-PublishedTemplates.ps1

```diff
@@ -0,0 +1,62 @@
+function Get-PublishedTemplates {
+    <#
+    .SYNOPSIS
+    Get published certificate templates from Active Directory.
+
+    .DESCRIPTION
+    Gets all templates from Active Directory and identifies which ones are published.
+
+    .EXAMPLE
+    Get-PublishedTemplates
+
+    .NOTES
+        If either of these flags are set, the template is considered published:
```

Comment:
Your logic doesn't seem consistent with what's actually published, but it's very possible I'm missing something.

Your code returns this from my lab CA:
<img width="809" alt="image" src="https://github.com/user-attachments/assets/cf82f98b-ac32-4215-afb9-f457297dcccf">

But the Certificate Templates list on my lab CA looks like this:
<img width="753" alt="image" src="https://github.com/user-attachments/assets/a8b16bba-7977-4fa4-9eb9-ccb708564030">

Which matches CA's object's `cetificateTemplates` attribute expanded:
<img width="1164" alt="image" src="https://github.com/user-attachments/assets/eb3e3101-f9f2-41c3-a0c7-cf7e1b9e6c6b">

Note: The Certificate Templates pane shows the templates `displayName` attribute while the expanded `certificateTemplates` list shows the `name` attribute.

This function should be rewritten to see if the template name exists within the `certificateTemplates` attribute on any `pKIEnrollmentService` object. We already collect the required attribute in `Get-ADCSObject`, so I think you could:

1. Collect all the published template `name`s from all the CA objects
2. De-duplicate the list and store it as `$PublishedTemplates`
3. Loop through the templates to see if `$PublishedTemplates -contains $TemplateName`
4. Return `$true` or `$false` to enrich the template object as a custom attribute.

Or whatever you want to do. You probably have a more elegant way of doing it.

Reply:
Thinking about de-duplicating the list and the possibility of different versions of a template being published on different CAs. Is that possible, and if so: does a published template on multiple CAs have the same OID for each instance, or a unique one per CA?